src/main.py

```diff
@@ -72,3 +77,10 @@
     # get top k terms with the highest idf score (this is atm not very useful
     # but it will be used for the tf.idf similarity stuff
     top_keywords = get_top_idf_features(train_description_feature, 5)
+
+    # calculate similarity between train an testset job descriptions
+    # this is of high order complexity - test it on a subset of the data
+    corpus_list = pandas_vector_to_list(train_description_feature)
+    queries_list = pandas_vector_to_list(test_description_feature)
+    similarity = get_tfidf_similarity(corpus_list, queries_list)
+    relevant_documents = extract_relevant_documents(similarity)
```

Comment:
For people who don't want to bother with running the code: 
similarity is a (size(train)  size(test)) matrix that has the correlation atings for record to each record.
relevant_documents is the filtered version that is a dictionary, holds any relationship that is more significant than 0.8
Key: 2 value(10, 0.9888 ), meaning second and tenth record's description has huge correlation

Which brings me to the question: @blanche How do you intent to feed this to a machine learning algorithm ? Relevant documents hold the correlation for many to many records and can't be fed as a single feature(column), the simplest way of feeding it is, you have to separate them as:

Correlation_relation_record_to_2 where record is each record(row) and 2 is compared record, but if we use this method we will have 280k features which will blow up the feature set. ( This is identical to adding similarity matrix directly to set )

I get that some descriptions look like other and it is cool that we find it out but I would prefer if you come up with a solution before merging this.
_____________
Some examples from results:
6-26 : Senior Process engineer - junior process engineer
8-109: Chef De Partie needed - Chef De Partie needed, this is literally the same job from same recruiter for same position in same city
20-56: Title Software Engineer - Software Engineer

If the descriptions are almost the same that means the job titles and job category is (almost) the same ( which is). Adding yet another feature that says " hey look these records are similar to each other " will probably do unnecessary emphasis on job titles more than it should.
_____________

TLDR: Nice work, but me wants to see the result in a way we can feed in to ML.